File: dsa-problems/Sorting/solutions/java/SRT-012-count-within-threshold-after-self.java
import java.util.*;

class Solution {
    public long[] countWithinThreshold(int[] arr, long T) {
        int n = arr.length;
        long[] counts = new long[n];

        for (int i = 0; i < n; i++) {
            for (int j = i + 1; j < n; j++) {
                if ((long)arr[j] - (long)arr[i] <= T) {
                    counts[i]++;
                }
            }
        }

        return counts;
    }
}

class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        if (!sc.hasNextInt()) {
            sc.close();
            return;
        }
        int n = sc.nextInt();
        long t = sc.nextLong();
        int[] arr = new int[n];
        for (int i = 0; i < n; i++) {
            arr[i] = sc.nextInt();
        }
        Solution solution = new Solution();
        long[] result = solution.countWithinThreshold(arr, t);
        StringBuilder sb = new StringBuilder();
        for (int i = 0; i < result.length; i++) {
            if (i > 0) sb.append(' ');
            sb.append(result[i]);
        }
        System.out.println(sb.toString());
        sc.close();
    }
}

File: dsa-problems/Sorting/solutions/cpp/SRT-012-count-within-threshold-after-self.cpp
#include <vector>
#include <iostream>

using namespace std;

class Solution {
public:
    vector<long long> countWithinThreshold(const vector<int>& arr, long long T) {
        int n = arr.size();
        vector<long long> counts(n, 0);

        for (int i = 0; i < n; i++) {
            for (int j = i + 1; j < n; j++) {
                if ((long long)arr[j] - (long long)arr[i] <= T) {
                    counts[i]++;
                }
            }
        }

        return counts;
    }
};

int main() {
    ios::sync_with_stdio(false);
    cin.tie(nullptr);

    int n;
    long long t;
    if (!(cin >> n >> t)) return 0;
    vector<int> arr(n);
    for (int i = 0; i < n; i++) {
        cin >> arr[i];
    }
    Solution solution;
    vector<long long> result = solution.countWithinThreshold(arr, t);
    for (int i = 0; i < (int)result.size(); i++) {
        if (i) cout << ' ';
        cout << result[i];
    }
    cout << "\n";
    return 0;
}
